pennylane/pauli/conversion.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions to convert between ``~.PauliSentence`` and other PennyLane operators.
+"""
+from operator import matmul
+from itertools import product
+from typing import Union
+from functools import reduce
+
+import numpy as np
+
+from pennylane.ops import Hamiltonian
+from .pauli_arithmetic import PauliWord, PauliSentence, I, X, Y, Z, mat_map, op_map
+
+
+def decompose(
```

Comment:
Realizing that `decompose`is actually a slightly confusing name.

How about we have `hamiltonian_from_matrix` and `paulisentence_from_matrix`?

A good rule of thumb is "the more narrow the use case, the longer the name".  I think this should have a fully descriptive function name.

Reply:
I don't think we need two functions, especially since we will be deprecating one, once we deprecate hamiltonians.